### PR TITLE
Add support for `pd.Series`

### PR DESCRIPTION
**Pull request by rsnirwan**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/dataset/dataframe.py [hunk 1]
```diff
@@ -175,6 +182,33 @@ def from_long_dataframe(
...
+    def to_df(series):
+        assert isinstance(
+            series.index, DatetimeIndexOpsMixin
+        ), "series index has to be a DatetimeIndex."
+        return series.to_frame(name="target")
+
+    if isinstance(series, list):
+        return list(map(to_df, series))
+    elif isinstance(series, dict):
+        return {k: to_df(s) for k, s in series.items()}
```
Comment (jaheba) [inline src/gluonts/dataset/dataframe.py]: Please let us try not having single-letter variable names.

Comment (jaheba) [reply src/gluonts/dataset/dataframe.py]: Also, in this case you can use `toolz.valmap`

File: src/gluonts/dataset/dataframe.py [hunk 2]
```diff
@@ -175,6 +182,33 @@ def from_long_dataframe(
...
+
+
+def is_series(series: Any) -> bool:
+    """
+    return True if `series` is `pd.Series` or a collection of `pd.Series`.
+    """
+    if isinstance(series, list):
+        return isinstance(series[0], pd.Series)
+    elif isinstance(series, dict):
+        return isinstance(list(series.values())[0], pd.Series)
```
Comment (jaheba) [inline src/gluonts/dataset/dataframe.py]:
```suggestion
        return is_series(list(series.values()))
```

File: src/gluonts/dataset/dataframe.py [hunk 3]
```diff
@@ -175,6 +182,33 @@ def from_long_dataframe(
...
+        return {k: to_df(s) for k, s in series.items()}
+    return to_df(series)
+
+
+def is_series(series: Any) -> bool:
+    """
+    return True if `series` is `pd.Series` or a collection of `pd.Series`.
+    """
+    if isinstance(series, list):
+        return isinstance(series[0], pd.Series)
```
Comment (jaheba) [inline src/gluonts/dataset/dataframe.py]:
```suggestion
        return is_series(series[0])
```

File: src/gluonts/dataset/dataframe.py [hunk 4]
```diff
@@ -175,6 +182,33 @@ def from_long_dataframe(
...
+        return series.to_frame(name="target")
+
+    if isinstance(series, list):
+        return list(map(to_df, series))
+    elif isinstance(series, dict):
+        return {k: to_df(s) for k, s in series.items()}
+    return to_df(series)
+
+
+def is_series(series: Any) -> bool:
```
Comment (jaheba) [inline src/gluonts/dataset/dataframe.py:202]: This looks a case for `TypeGuard`

Comment (rsnirwan) [reply src/gluonts/dataset/dataframe.py:202]: Thanks for the comments @jaheba ! 
I didn't get this one. Do you mean this: https://typeguard.readthedocs.io/ ?
I wouldn't know who to use `TypeGuard` here. I am using this function for the dispatch to `series_to_dataframe` only, not to check for wrong types.

Comment (jaheba) [reply src/gluonts/dataset/dataframe.py:202]: I was referring to: https://peps.python.org/pep-0647/

But maybe it's actually not that useful here.

**Review by jaheba — APPROVED** [review]

**Pull request merged**